### PR TITLE
FOPTS-3684 Azure Synapse SQL Pools

### PR DESCRIPTION
**Pull request by Rangelmv23**
### Description

Policy to get optimization recommendations for Azure Synapse service to realize more savings.

### Issues Resolved

Policy can recommend rightsizing dedicated pool compute resources or recommend to pause them to result in substantial potential savings.

### Link to Example Applied Policy

https://app.flexeratest.com/orgs/1105/automation/applied-policies/projects/104308?policyId=667dd350067ffcbfacc52699

### Contribution Check List

- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/rightsize_synapse_sql_pools/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- The policy identifies all instances that have metrics below the user-specified thresholds and provides the relevant recommendation.
+- The recommendation provided for idle instances (no connections) is a pause action. These instances can be paused in an automated manner or after approval.
+- The Recommendations to downsize underutilized instances are also generated for review and approval. 
+
+### Policy Savings Details
+
+The policy includes the estimated monthly savings. The estimated monthly savings is recognized for idle resources if the resource is paused, and for underutilized resources if the resource is downsized.
+
+- The `Estimated Monthly Savings` is calculated by multiplying the amortized cost of the resource for 1 day by 30.44, which is the average number of days in a month.
+- For idle resources, the savings is the full cost of the resource. For underutilized resources, the savings is the difference between the current cost and the cost after downsizing.
```
Comment (midir99) [inline cost/azure/rightsize_synapse_sql_pools/README.md:19]:
```suggestion
- For idle resources, the savings are the full cost of the resource. For underutilized resources, the savings is the difference between the current cost and the cost after downsizing.
```

File: cost/azure/rightsize_synapse_sql_pools/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- The incident message detail includes the sum of each resource `Estimated Monthly Savings` as `Potential Monthly Savings`.
```
Comment (midir99) [inline cost/azure/rightsize_synapse_sql_pools/README.md:20]:
```suggestion
- The incident message details includes the sum of each resource `Estimated Monthly Savings` as `Potential Monthly Savings`.
```

File: cost/azure/rightsize_synapse_sql_pools/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- *Allow/Deny Subscriptions List* - A list of allowed or denied Subscription IDs/names. If empty, no filtering will occur and recommendations will be produced for all subscriptions.
+- *Allow/Deny Regions* - Whether to treat Allow/Deny Regions List parameter as an allow or deny list. Has no effect if Allow/Deny Regions List is left empty.
+- *Allow/Deny Regions List* - Filter results by region, either only allowing this list or denying it depending on how the above parameter is set. Leave blank to consider all regions.
+- *Threshold* - Threshold to use when determining if a pool is underutilized.
+- *Statistic Lookback Period* - How many days back to look at metrics. This value cannot be set higher than 90 because Azure does not retain metrics for longer than 90 days.
+- *Minimum Time Active (Days)* - Only include pools in the results if they have not been paused for at least the number of specified days. Set to '0' to disable this filter and not consider minimum time active in the results.
```
Comment (midir99) [inline cost/azure/rightsize_synapse_sql_pools/README.md:34]:
```suggestion
- *Minimum Time Active (Days)* - Only include pools in the results if they have not been paused for at least the number of specified days. Set to '0' to disable this filter and not consider the minimum time active in the results.
```

File: cost/azure/rightsize_synapse_sql_pools/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+The [Provider-Specific Credentials](https://docs.flexera.com/flexera/EN/Automation/ProviderCredentials.htm) page in the docs has detailed instructions for setting up Credentials for the most common providers.
+
+## Supported Clouds
+
+- Azure
+
+## Cost
+
+This Policy Template does not incur any cloud costs
```
Comment (midir99) [inline cost/azure/rightsize_synapse_sql_pools/README.md:72]:
```suggestion
This Policy Template does not incur any cloud costs.
```

**Review by midir99 — APPROVED** [review]
I ran Grammarly to check for grammar issues, anyway I don't find some of these really important but I'm not an expert writing English.

**Review by night-codes — APPROVED** [review]

**Pull request merged**